src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (head == null) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            Node<T> node = new Node<>(null, value, head);
+            head = node;
+            if (size == 0) {
+                tail = node;
+            }
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else {
```

Comment:
to make it more readable, extract this case to the top of the method and call return after add. Then the rest of the code won't be indented and thus easier to read